### ISSUE TITLE
User vault-injector cannot patch resource "mutatingwebhookconfigurations" in API group "admissionregistration.k8s.io" at the cluster scope

### ISSUE THREAD

**Issue by DrissiReda** [hashicorp/vault-k8s#30]
I am trying to deploy the vault injector locally. I did 2 changes: 
- I use HTTP so I replaced all HTTPS by HTTP on yaml files under develop, and 443 ports by 80.
- I use the "default" namespace instead of the "vault" namespace.

All I did afterwards was modify the vault url by my vault's url. And used kubectl to create objects described in the 4 injector yaml files under develop.

This is the output of the vault-injector-XXXXXXX pod
```bash
Listening on ":8080"...
2020-01-03T13:50:17.146Z [INFO]  handler: Starting handler..
Updated certificate bundle received. Updating certs...
Error updating MutatingWebhookConfiguration: mutatingwebhookconfigurations.admissionregistration.k8s.io "vault-agent-injector-cfg" is forbidden: User "system:serviceaccount:default:vault-injector" cannot patch resource "mutatingwebhookconfigurations" in API group "admissionregistration.k8s.io" at the cluster scope
Error updating MutatingWebhookConfiguration: mutatingwebhookconfigurations.admissionregistration.k8s.io "vault-agent-injector-cfg" is forbidden: User "system:serviceaccount:default:vault-injector" cannot patch resource "mutatingwebhookconfigurations" in API group "admissionregistration.k8s.io" at the cluster scope
Error updating MutatingWebhookConfiguration: mutatingwebhookconfigurations.admissionregistration.k8s.io "vault-agent-injector-cfg" is forbidden: User "system:serviceaccount:default:vault-injector" cannot patch resource "mutatingwebhookconfigurations" in API group "admissionregistration.k8s.io" at the cluster scope
```

Anyone knows where this permission problem comes from?

**Comment by jasonodonnell** [comment]
It sounds like you didn't bind the cluster role to your service account to give it permissions for admission webhooks: https://github.com/hashicorp/vault-k8s/blob/master/deploy/injector-rbac.yaml

Did you get an error when trying to create this?  Do you have permissions to create cluster roles?

**Comment by DrissiReda** [comment]
I didn't get any error with that. I should have permissions to create anything.

**Comment by jasonodonnell** [comment]
@DrissiReda, did you update this value with your [namespace](https://github.com/hashicorp/vault-k8s/blob/master/deploy/injector-rbac.yaml#L40)?

**Comment by DrissiReda** [comment]
Yes I have. I gave up on deploying this on the default namespace, it was never a good idea to begin with

**Comment by jasonodonnell** [comment]
@DrissiReda After reviewing this again, I noticed you said this:

> I use HTTP so I replaced all HTTPS by HTTP on yaml files under develop, and 443 ports by 80.

This is _not_ recommended for the injector.  Kubernetes requires that admission controllers be HTTPS enabled.  The agent injector will automatically setup TLS for you, so there's no reason to not use it.  I would suggest switching back to HTTPS, it might be what's causing issues here.

**Comment by Kampe** [comment]
I see this same issue with the default namespace, I apply the manifest with kustomize without any overlays - not sure what the issue is.

**Comment by tiagovrtr** [comment]
I'm having the same issue while running the injector in the default namespace

**Comment by saurabh-mish** [comment]
Seeing the same error while updating a deployment YAML in a non-default namespace
